sql_generators/glean_usage/common.py

```diff
@@ -104,7 +104,14 @@ def table_names_from_baseline(baseline_table, include_project_id=True):
 def referenced_table_exists(view_sql):
     """Dry run the given view SQL to see if its referent exists."""
     dryrun = DryRun("foo/bar/view.sql", content=view_sql)
-    return 404 not in [e.get("code") for e in dryrun.errors()]
+    # 403 is returned if referenced dataset doesn't exist; we need to check that the 403 is due to dataset not existing
+    # since dryruns on views will also return 403 due to the table CREATE
+    # 404 is returned if referenced table or view doesn't exist
+    return not any([
+        404 == e.get("code") or 
+        (403 == e.get("code") and "does not exist in location" in e.get("message")) 
+        for e in dryrun.errors()
+    ])
```

Comment:
- Relying on specific error messages seems a bit tenuous either way (e.g. they could arbitrarily change "does not" to "doesn't"), but I think excluding the unwanted error would be slightly safer.
- IMO boolean operators at the beginning of lines are better for code readability.
```suggestion
    return not any([
        404 == e.get("code")
        or (403 == e.get("code") and "bigquery.tables.create denied" not in e.get("message")) 
        for e in dryrun.errors()
    ])
```